Sources/XMLParser/XMLParser.swift

```diff
@@ -58,6 +58,7 @@ let commentParser = ParsePrint {
 }
 
 let textParser = ParsePrint {
+    "".utf8 // noop. required to force Whitespace below to take the correct buildExpression
```

Comment:
Can't you add a generic to the Whitespace itself? Or is that not what this is fixing? Or perhaps at the ParsePrint level maybe?

Reply:
You're right, I added the generic specialisation instead of the extra line.

Reply:
actually used the more recommended way to deal with this cleanly.